fenics_helpers/timestepping.py

```diff
@@ -55,25 +55,37 @@ def __init__(self, solve, post_process, u):
         self._post_process = post_process
         self._u = u
 
-    def run(self, t_end, t_start=0.0, dt0=None, show_bar=False):
+    def run(self, t_end, t_start=0.0, dt0=None, checkpoints=None, show_bar=False):
         if dt0 is None:
             dt0 = self.dt_max
 
+        if checkpoints is not None:
+            checkpoints = np.sort(checkpoints)
+            current_checkpoint = 0
+
         dt = dt0
 
         u_prev = self._u.copy(deepcopy=True)
         t = t_start
 
         progress = Progress(t_start, t_end, show_bar)
         while t < t_end:
-            t += dt
+            if checkpoints is not None and checkpoints[current_checkpoint] < t + dt:
```

Comment:
What is the expected behavior for `t_start = 0` and `checkpoints=[-1]`? 
Currently, this would mean starting at `t = -1`, which is bad IMO.

I would expect that the time stepping includes the checkpoints. So silently excluding them, is also bad.

What do think about adding 
~~~py
assert(checkpoints[0] >= t_start)
assert(checkpoints[-1] <= t_end)
~~~
after `np.sort`?